reconcile/openshift_resources_base.py

```diff
@@ -349,6 +349,17 @@ def urlunescape(string: str, encoding: Optional[str] = None) -> str:
     return parse.unquote(string, encoding=encoding)
 
 
+def hash_list(input):
+    lst = list(input)
+    sorted_list = sorted(lst, key=lambda x: x["name"])
```

Comment:
this code expects the list elements to be dicts that have at least a name key. isn't that restricting the reusability of the hash_list filter?

Reply:
You are absolutely right! I need to add more args to the filter to make it reusable. Also the char length of the hash.
This is still very early draft mode - will adjust the PR description accordingly. Thanks though for the fast input :) :bow:

Reply:
Probably a filter is not the best idea in the first place here - might be better to do what we do for `query()` too. That way we have a function with arguments we can pass, like

```jinja2
{% hash_list(namespaces, length=6, sort_by=lambda x: x["name"]) %}
```

Reply:
I came to the conclusion that lambda wont cut it. It would force us to do some kind of `eval()` on strings defined in jinja2 templates. I dont like that at all for security reasons. So I expect now a flat list of non-dicts here. A use-case can be found here: https://gitlab.cee.redhat.com/service/app-interface/-/blob/0800c7be8f25dfff09e9c22acd79c7480bc7f673/resources/setup/dynatrace/log-config-job.yaml.j2#L8